contracts/eosio.system/src/eosio.system.cpp

```diff
@@ -397,6 +398,67 @@ namespace eosiosystem {
       set_resource_limits( account, current_ram, current_net, cpu );
    }
 
+   void system_contract::denynames( const std::vector<name>& patterns ) {
+      require_auth( get_self() );
+
+      if (patterns.empty())
+         return; // no-op for empty list, consistent with ignoring duplicate names.
+
+      check(patterns.size() <= 512, "Cannot provide more than 512 patterns in one action call");
+
+      account_name_blacklist_table bl_table(get_self(), get_self().value);
+      auto itr = bl_table.begin();
+      bool present = (itr != bl_table.end());
+
+      auto add_patterns_to = [&patterns](auto& current) {
+         // number of blackcklist name patterns expected to be small, so quadratic check OK
+         for (auto n : patterns) {
+            // check that pattern is valid (pattern should not be empty or more than 12 character long)
+            canon_name_t pattern(n);
+            if (!pattern.valid())
+               continue; // silently ignore incorrect patterns
```

Comment:
I would rather that we abort the action with an error message if there is an invalid pattern in the list.

Reply:
https://github.com/eosnetworkfoundation/eos-system-contracts/pull/189/commits/8aaef04c6c09baca2490c02dd9b08ae0c75df0aa